uaclient/api/u/pro/config/v1.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Optional
+
+from uaclient import util
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.config import UA_CONFIGURABLE_KEYS, UAConfig
+from uaclient.data_types import (
+    BoolDataValue,
+    DataObject,
+    Field,
+    IntDataValue,
+    StringDataValue,
+)
+from uaclient.files import user_config_file
+
+LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
+
+
+class ConfigInfo(DataObject, AdditionalInfo):
+    fields = [
+        Field("http_proxy", StringDataValue, required=False, doc="HTTP proxy"),
+        Field(
+            "https_proxy", StringDataValue, required=False, doc="HTTPS proxy"
+        ),
+        Field(
+            "ua_apt_http_proxy",
+            StringDataValue,
+            required=False,
+            doc="UA APT HTTP proxy",
+        ),
+        Field(
+            "ua_apt_https_proxy",
+            StringDataValue,
+            required=False,
+            doc="UA APT HTTPS proxy",
+        ),
+        Field(
+            "global_apt_http_proxy",
+            StringDataValue,
+            required=False,
+            doc="Global APT HTTP proxy",
+        ),
+        Field(
+            "global_apt_https_proxy",
+            StringDataValue,
+            required=False,
+            doc="Global APT HTTPS proxy",
+        ),
+        Field("apt_news", BoolDataValue, required=False, doc="APT news"),
+        Field(
+            "apt_news_url", StringDataValue, required=False, doc="APT news URL"
+        ),
+        Field(
+            "metering_timer",
+            IntDataValue,
+            required=False,
+            doc="Metering timer",
+        ),
+        Field(
+            "update_messaging_timer",
+            IntDataValue,
+            required=False,
+            doc="Update messaging timer",
+        ),
+    ]
+
+    def __init__(
+        self,
+        *,
+        http_proxy: Optional[str] = None,
+        https_proxy: Optional[str] = None,
```

Comment:
I think it is worth having integration tests where we set proxy values and see how they behave on `non-root` vs `root` users

Reply:
Yes, makes sense. Will add those to check if the config output is correct in both cases